src/Evolution/DiscontinuousGalerkin/Limiters/WenoOscillationIndicator.cpp

```diff
@@ -186,32 +236,19 @@ double oscillation_indicator(const DerivativeWeight derivative_weight,
                              const DataVector& data,
                              const Mesh<VolumeDim>& mesh) noexcept {
   ASSERT(mesh.basis() == make_array<VolumeDim>(Spectral::Basis::Legendre),
-         "No implementation for mesh: " << mesh);
-
-  // Optimization: cache the indicator matrix, because it does not depend on the
-  // data. This caching will have to be generalized to handle simulations with
-  // more than one mesh.
-  const auto cache = make_static_cache<CacheEnumeration<
-      DerivativeWeight, DerivativeWeight::Unity, DerivativeWeight::PowTwoEll,
-      DerivativeWeight::PowTwoEllOverEllFactorial>>(
-      [&mesh](const DerivativeWeight dw) noexcept -> Matrix {
-        return compute_indicator_matrix(dw, mesh);
-      });
-  const Matrix indicator_matrix = cache(derivative_weight);
-
-  // Check there is only one mesh, i.e., that we satisfy the cache assumptions.
-  static const Mesh<VolumeDim> mesh_for_cached_matrix = mesh;
-  ASSERT(
-      mesh_for_cached_matrix == mesh,
-      "This call to oscillation_indicator received as argument a different\n"
-      "Mesh than was previously cached, suggesting that multiple meshes are\n"
-      "used in the computational domain. This is not (yet) supported.\n"
-      "Cached mesh: "
-          << mesh_for_cached_matrix
-          << "\n"
-             "Argument mesh: "
-          << mesh);
+         "Unsupported mesh: " << mesh);
```

Comment:
This only asserts the basis, so add something like "This function assumes a Legendre basis"?